dev_tools/packaging/verify-published-package.sh

```diff
@@ -0,0 +1,97 @@
+# limitations under the License.
+
+################################################################################
+# Downloads and tests openfermion wheels from the pypi package repository. Uses the
+# prod pypi repository unless the --test switch is added.
+#
+# CAUTION: when targeting the test pypi repository, this script assumes that the
+# local version of openfermion has the same dependencies as the remote one (because the
+# dependencies must be installed from the non-test pypi repository). If the
+# dependencies disagree, the tests can spuriously fail.
+#
+# Usage:
+#     dev_tools/packaging/verify-published-package.sh PACKAGE_VERSION [--test|--prod]
+################################################################################
+
+set -e
+trap "{ echo -e '\e[31mFAILED\e[0m'; }" ERR
+
+
+PROJECT_NAME=openfermion
+PROJECT_VERSION=$1
+PROD_SWITCH=$2
+
+if [ -z "${PROJECT_VERSION}" ]; then
+    echo -e "\e[31mFirst argument must be the package version to test.\e[0m"
+    exit 1
+fi
+
+if [ "${PROD_SWITCH}" = "--test" ]; then
+    PYPI_REPOSITORY_FLAG="--index-url=https://test.pypi.org/simple/"
+    PYPI_REPO_NAME="TEST"
+elif [ -z "${PROD_SWITCH}" ] || [ "${PROD_SWITCH}" = "--prod" ]; then
+    PYPI_REPOSITORY_FLAG=''
+    PYPI_REPO_NAME="PROD"
+else
+    echo -e "\e[31mSecond argument must be empty, '--prod' or '--test'.\e[0m"
+    exit 1
+fi
+
+# Find the repo root.
+cd "$( dirname "${BASH_SOURCE[0]}" )"
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+
+# Temporary workspace.
+tmp_dir=$(mktemp -d "/tmp/verify-published-package.XXXXXXXXXXXXXXXX")
+cd "${tmp_dir}"
+trap "{ rm -rf ${tmp_dir}; }" EXIT
+
+# Test both the python 2 and python 3 versions.
+for PYTHON_VERSION in python2 python3; do
+    # Prepare.
+    RUNTIME_DEPS_FILE="${REPO_ROOT}/requirements.txt"
+    echo -e "\n\e[32m${PYTHON_VERSION}\e[0m"
+    echo "Working in a fresh virtualenv at ${tmp_dir}/${PYTHON_VERSION}"
+    virtualenv --quiet "--python=/usr/bin/${PYTHON_VERSION}" "${tmp_dir}/${PYTHON_VERSION}"
+
+    # Install package.
+    if [ "${PYPI_REPO_NAME}" == "TEST" ]; then
+        echo "Pre-installing dependencies since they don't all exist in TEST pypi"
+        "${tmp_dir}/${PYTHON_VERSION}/bin/pip" install --quiet -r "${RUNTIME_DEPS_FILE}"
+    fi
+    echo Installing "${PROJECT_NAME}==${PROJECT_VERSION} from ${PYPI_REPO_NAME} pypi"
+    "${tmp_dir}/${PYTHON_VERSION}/bin/pip" install --quiet ${PYPI_REPOSITORY_FLAG} "${PROJECT_NAME}==${PROJECT_VERSION}"
+
+    # Check that code runs without dev deps.
+    echo Checking that code executes
+    "${tmp_dir}/${PYTHON_VERSION}/bin/python" -c "import openfermion; print(openfermion.FermionOperator((1, 1)))"
+    "${tmp_dir}/${PYTHON_VERSION}/bin/python" -c "import openfermion; print(openfermion.QubitOperator((1, 'X')))"
+
+    # Run tests.
+    echo Installing pytest requirements
+    if [ "${PYTHON_VERSION}" = "python2" ]; then
+        "${tmp_dir}/${PYTHON_VERSION}/bin/pip" install --quiet pytest mock
```

Comment:
Are these requirements the same for openfermion as they are for cirq and openfermioncirq? You probably don't need `mock`.

Just unconditionally do `"${tmp_dir}/${PYTHON_VERSION}/bin/pip" install --quiet pytest mock` regardless of version